kodi_addon_checker/__main__.py

```diff
@@ -66,7 +67,7 @@ def main():
         for directory in args.dir:
             report.add(check_artifact(directory, args))
```

Comment:
This should also honor the branch argument, shouldn't it?

Reply:
yes and actually I have also missed to add another argument in `check_artifact` function.

 But what I am now finding contradictory is that if we pass single addon  [this](https://github.com/xbmc/addon-check/blob/master/kodi_addon_checker/__main__.py#L41) will directly call the start function. But we are getting [`repo_addons`](https://github.com/xbmc/addon-check/pull/63/files#diff-87cbf8a463aa7f67c8382bdac08ec809R15) from `check_repo` function. So I'll have to do this within the start function but if I do that then we'll have the same issue of having multiple requests for same repo i.e caching addon.xml.

@Razzeee Am I on the right track or have I missed something here ?

Reply:
That evaluation sounds right.

Reply:
I am currently trying to figure it out but I am open to suggestions :wink:

Reply:
do a preparation step in __main__.py where you get everything that we want to keep in that overarching scope?
Alternative might be to even write that to the disk, not sure I'm a fan of that.

Reply:
We'll you can always add another class between those. If theres a good reason to do this.